src/directives/bindInput.ts

```diff
@@ -227,20 +236,29 @@ class BindInputDirective extends AsyncDirective {
   private __getValueFromElement(element: Element): unknown {
     let value: unknown = undefined;
 
-    if (isInputElement(element)) {
-      if (element.type === 'checkbox') {
-        value = element.checked === true;
-      } else {
-        value = element.value;
+    if (isSelectElement(element)) {
+      switch (element.type.toLowerCase()) {
+        case 'select-multiple':
+          value = [...element.selectedOptions].map((opt) => opt.value);
+          break;
+        case 'select-one':
+        default:
+          value = element.value;
       }
-    } else if (isSelectElement(element)) {
-      if (element.multiple) {
-        value = [...element.selectedOptions].map((opt) => opt.value);
-      } else {
-        value = element.value;
+    } else if (isInputElement(element)) {
+      switch (element.type.toLowerCase()) {
+        case 'checkbox':
+          value = element.checked === true;
+          break;
+        case 'number':
+        case 'date':
+        case 'time':
+          value = element.valueAsNumber;
```

Comment:
date/time are an odd one

i wonder if we should be exposing it as a `Date` (via `valueAsDate`), or if not, keep it as a string. a number seems unexpected since `value` will be a date/time string on the native input

so i think it should be a string or a date but not a number